src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,108 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int INITIAL_CAPACITY = 16;
+    private static final double LOAD_FACTOR = 0.75;
+    private int threshold;
+    private int size;
+    private Node<K,V>[] table;
+
+    public MyHashMap() {
+        threshold = (int) (INITIAL_CAPACITY * LOAD_FACTOR);
+        table = (Node<K, V>[]) new Node[INITIAL_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
+        if (size > threshold) {
+            resize();
+        }
+        putNewValue(key, value);
+    }
 
+    private void putNewValue(K key, V value) {
+        int nodeIndex = getIndexForNode(key);
+        Node<K, V> nodeToPut = new Node<>(key, value);
+        if (table[nodeIndex] == null) {
+            table[nodeIndex] = nodeToPut;
+        } else {
+            Node<K, V> currentNode = table[nodeIndex];
+            while (Objects.equals(currentNode.key, key) || currentNode.next != null) {
+                if (!Objects.equals(currentNode.key, key)) {
+                    currentNode = currentNode.next;
+                } else {
+                    currentNode.value = nodeToPut.value;
+                    return;
```

Comment:
```suggestion
                if (Objects.equals(currentNode.key, key)) {
                    currentNode.value = nodeToPut.value;
                    return;
                }
                   currentNode = currentNode.next;
```
this way we don't need else

Reply:
Makes great sense, thank you) refactored.